src/chapters/orm-intro/background.rst

```diff
@@ -0,0 +1,170 @@
+
+Install EF Core Tools
+#####################
+
+In this section, "terminal" refers to the Terminal app in MacOS and Powershell in Windows (use *Tools > Command Line > Developer Powershell* to open).
+
+Open a terminal and run:
+
+.. sourcecode:: bash
+
+   $ dotnet tool install -g dotnet-ef
+
+This command installs a set of command-line tools for working with EntityFrameworkCore *globally*, which means it will be available for any ASP.NET project we use in the future. We will use the tools provided by this package to update our database schema after adding or changing model classes. 
+
+.. note::
+
+   This not applies to *MacOS users only*.
+
+   For these tools to be accessible from the command line, they must be within your user path. Open ``~/.bash_profile`` and add the following line to the very bottom (recall that ``~`` is shorthand for your home directory, which is the directory you are in when you open a new terminal).
+
+   .. sourcecode:: bash
+
+      export PATH="$PATH:$HOME/.dotnet/tools/"
+
+   This will append the location of the EF tools to your user path. Save and close the file. Then close your terminal window and open a new one, so that the changes can take effect.
+
+To test that this install worked, run ``dotnet ef``. The output should be a message displaying basic EF tool commands and options.
+
+Once you have taken these steps, you are ready to set up the appropriate models and controllers for the application. We'll do that in the next section.
+
+.. index:: ! environment variables
+
+Key Takeaways
+^^^^^^^^^^^^^
+
+Before we can get into the ins and outs of using ORM, we need to make sure that our application has a corresponding database and that our application is ready to connect to MySQL. We can start to do this by creating new schemas and setting user privileges in MySQL Workbench. We also *must* make sure that the MVC application has the correct dependencies, username, and password to access the schema.
+
+If we do not do these steps, then our application will not be able to use a persistent data source.
+
+As we noted in the video, while we can simply set the value of the ``DefaultConnection`` property using the values of the username and password, this is NOT best practice. We regularly commit our code to Github, meaning anyone who reads the code in our repository can see the username and password. While you can do it for the applications in this class, you do not want to do it in the future.
+
+To avoid this in the future, you can configure your ``DefaultConnection`` string to reference **environment variables**.
+You then hide the appropriate info by setting the environment variable's value equal to the password, for example.
+
+Microsoft has `documentation <https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1#environment-variables>`_ on how to configure your environment variables to keep the username and password to your database safe and secure.
```

Comment:
how would feel putting this note into an admonition? or maybe just the portion starting with "to avoid this in the future..."